From 812098cb7e971724203091cd8746dc7d1969c23f Mon Sep 17 00:00:00 2001
From: Karmjit Mahil <karmjit.mahil@igalia.com>
Date: Mon, 14 Sep 2026 12:49:28 +0100
Subject: [PATCH] tu: Fix inline uniform update offset

As of 37cde2c6348 ("tu: Rewrite inline uniform implementation") the
indirection was removed, so the UBO descriptor which we were skipping
was removed. Fix a leftover descriptor skip.

Assisted-by: OpenCode (DeepSeek V4 Flash 0731)
Signed-off-by: Karmjit Mahil <karmjit.mahil@igalia.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/44429>
(cherry picked from mesa main commit 812098cb7e971724203091cd8746dc7d1969c23f)
---
 src/freedreno/vulkan/tu_descriptor_set.cc | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/freedreno/vulkan/tu_descriptor_set.cc b/src/freedreno/vulkan/tu_descriptor_set.cc
index 41763121436..22f49e8dfec 100644
--- a/src/freedreno/vulkan/tu_descriptor_set.cc
+++ b/src/freedreno/vulkan/tu_descriptor_set.cc
@@ -1554,15 +1554,15 @@ tu_update_descriptor_sets(const struct tu_device *device,
             if (src_remaining == 0) {
                src_binding_layout++;
                src_ptr = src_set->mapped_ptr + src_binding_layout->offset / 4;
-               src = (uint8_t *)(src_ptr + FDL6_TEX_CONST_DWORDS);
-               src_remaining = src_binding_layout->size - 4 * FDL6_TEX_CONST_DWORDS;
+               src = (uint8_t *) src_ptr;
+               src_remaining = src_binding_layout->size;
             }
 
             if (dst_remaining == 0) {
                dst_binding_layout++;
                dst_ptr = dst_set->mapped_ptr + dst_binding_layout->offset / 4;
-               dst = (uint8_t *)(dst_ptr + FDL6_TEX_CONST_DWORDS);
-               dst_remaining = dst_binding_layout->size - 4 * FDL6_TEX_CONST_DWORDS;
+               dst = (uint8_t *) dst_ptr;
+               dst_remaining = dst_binding_layout->size;
             }
          } while (remaining > 0);
 
